src/agents/DexPoolScannerAgent.ts

```diff
@@ -0,0 +1,30 @@
+import {injectable} from 'inversify';
+
+import {LoopAgent} from './LoopAgent.js';
+import {BlockchainScanner} from 'src/services/BlockchainScanner.js';
+
+@injectable()
+export abstract class DexPoolScannerAgent extends LoopAgent {
+  abstract backoffTime: number;
+  abstract interval: number;
+  abstract stepInterval: number;
+  abstract dexPoolScanner: BlockchainScanner;
```

Comment:
```suggestion
  abstract dexPoolScanner: DexPoolScanner;
```

this service is for pool discovery, so let's rename to reflect the job.

Reply:
Good call. Done!